pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -594,15 +535,28 @@ func (m *cgroupManagerImpl) Update(cgroupConfig *CgroupConfig) error {
 	}
 
 	if unified {
-		if err := setResourcesV2(libcontainerCgroupConfig); err != nil {
-			return fmt.Errorf("failed to set resources for cgroup %v: %v", cgroupConfig.Name, err)
+		if err := propagateControllers(libcontainerCgroupConfig.Path); err != nil {
+			return err
```

Comment:
@odinuge why is this needed? runc's cgroup manager does the same thing in `Apply()` once, and doing it on every update/set seems excessive.

Reply:
> why is this needed?

I just want to keep it as close to the original code as possible to avoid regressions or new issues. I think we instead can improve it later when we have better test coverage, and can do more testing.

> runc's cgroup manager does the same thing in Apply() once, and doing it on every update/set seems excessive.

Yeah, I know. I can remove it tho., but I would instead like to do it in a follow up.

There is a lot of duplicate work in both k8s and libcontainer, especially when using systemd. I do think we can boost performance quite a bit by improving it (both in cpu cycles in kubelet and in locks taken by the kernel cgroup code), but that is outside the scope of this imo.

Reply:
Right; I missed it was in `setResourcesV2`. Can and should be addressed later.